assembly/decoder.ts

```diff
@@ -342,7 +345,7 @@ export class JSONDecoder<JSONHandlerT extends JSONHandler> {
       let charCode = this.readChar();
       numberAsString += String.fromCharCode(charCode);
 
-      if (charCode == CHAR_E || charCode == CHAR_E_LOWER || charCode == CHAR_PERIOD || charCode == CHAR_PLUS) {
+      if (charCode == CHAR_E || charCode == CHAR_E_LOWER || charCode == CHAR_PERIOD || charCode == CHAR_PLUS || charCode == CHAR_MINUS) {
```

Comment:
Hmm, but `+123` and `-123` could be integer as well. I guess better detect `e+` / `E+` and `e-` / `E-`